id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/helper/IdRepoServiceHelper.java

```diff
@@ -146,29 +143,43 @@ public String getSchema(String schemaVersion) {
         }
     }
 
-    public Map<String, HandleDto> getSelectedHandles(Object identity) throws IdRepoAppException {
-        Map<String, Object> requestMap = convertToMap(identity);
-        if (requestMap.containsKey(ROOT_PATH) && Objects.nonNull(requestMap.get(ROOT_PATH))) {
-            Map<String, Object> identityMap = (Map<String, Object>) requestMap.get(ROOT_PATH);
-            String schemaVersion = String.valueOf(identityMap.get(identityMapping.getIdentity().getIDSchemaVersion().getValue()));
-            String selectedHandlesFieldId = identityMapping.getIdentity().getSelectedHandles().getValue();
-            if(identityMap.containsKey(selectedHandlesFieldId) && Objects.nonNull(identityMap.get(selectedHandlesFieldId))) {
-                mosipLogger.debug(IdRepoSecurityManager.getUser(), ID_REPO_SERVICE_HELPER, "getSelectedHandles",
-                        requestMap.get(selectedHandlesFieldId));
-                List<String> selectedHandleFieldIds = (List<String>) identityMap.get(selectedHandlesFieldId);
-                return selectedHandleFieldIds.stream()
-                        .filter( handleFieldId -> supportedHandlesInSchema.get(schemaVersion).contains(handleFieldId))
-                        .collect(Collectors.toMap(handleName->handleName,
-                                handleFieldId-> {
-                                    String handle = ((String) identityMap.get(handleFieldId))
-                                            .concat(getHandlePostfix(handleFieldId))
-                                            .toLowerCase(Locale.ROOT);
-                                    return new HandleDto(handle, getHandleHash(handle));
-                                }));
-            }
-        }
-        return Map.of();
-    }
+	/**
+	 * @param identity it should contain the data for attributes coming as part of
+	 *                 selected handles.
+	 * @return handles map
+	 * @throws IdRepoAppException
+	 */
+	public Map<String, HandleDto> getSelectedHandles(Object identity) throws IdRepoAppException {
+		Map<String, Object> requestMap = convertToMap(identity);
+		if (requestMap.containsKey(ROOT_PATH) && Objects.nonNull(requestMap.get(ROOT_PATH))) {
+			Map<String, Object> identityMap = (Map<String, Object>) requestMap.get(ROOT_PATH);
+			String schemaVersion = String
+					.valueOf(identityMap.get(identityMapping.getIdentity().getIDSchemaVersion().getValue()));
+			String selectedHandlesFieldId = identityMapping.getIdentity().getSelectedHandles().getValue();
+			if (identityMap.containsKey(selectedHandlesFieldId)
+					&& Objects.nonNull(identityMap.get(selectedHandlesFieldId))) {
```

Comment:
Let's keep the if condition blocks smaller so that it's easy to read. We can reverse the condition here and return null right from this place.

Reply:
As discussed over the call, no need to change it as it is used by two if block conditions.